### PR TITLE
dep-update + added test for env size + print warning

### PR DESCRIPTION
**Pull request by trapacska**

### CODE REVIEW

File: main_test.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+			c:  Config{NewVersion: "0.0.1", ChangelogPath: "/tmp/changelog.md", WorkDir: "."},
+		},
+		{
+			cp: &mockTooManyChangelogPrinter{},
+			c:  Config{NewVersion: "0.0.1", ChangelogPath: "/tmp/changelog.md", WorkDir: "."},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			processChangelog(tt.cp, tt.c)
```
Comment (lpusok) [inline main_test.go:39]: Validation of processChangelog results could be done by checking that log.Warnf is called ot by returning a value if the size is too big.

File: main.go [hunk 2]
```diff
@@ -93,9 +83,40 @@ func main() {
...
+		failf("Failed to load envman configs, error: %s", err)
+	}
+
+	if envmanConfigs.EnvBytesLimitInKB > 0 {
+		envBytesLimit := envmanConfigs.EnvBytesLimitInKB * 1024
+
+		if len(changelog) > envBytesLimit {
+			log.Warnf("Changelog content exceeds the maximum allowed size to set in an environment variable. (%dKB)", envmanConfigs.EnvBytesLimitInKB)
+			log.Warnf("The changelog's content will be trimmed to fit the maximum allowed size.")
+			log.Warnf("It is possible to modify the limit by following this guide: https://devcenter.bitrise.io/tips-and-tricks/increasing-the-size-limit-of-env-vars")
```
Comment (lpusok) [inline main.go:97]: It would be helpful to reccomend using the changelog file to the user as an alternative.

File: main.go [hunk 3]
```diff
@@ -93,9 +83,40 @@ func main() {
...
+			changelog = changelog[:envBytesLimit-4] + "\n..."
+		}
+	}
+
 	if err := tools.ExportEnvironmentWithEnvman("BITRISE_CHANGELOG", changelog); err != nil {
```
Comment (lpusok) [inline main.go:102]: Would reccommend adding changelog_pth as an output, so users can see that there is an alternative.

Comment (trapacska) [reply main.go:102]: Changelog path is an input 🤔

Comment (lpusok) [reply main.go:102]: Adding it as an output makes it possible to be used in subsequent steps. The contents is an output of this step after all, so it makes sense I think as an output.

File: main.go [hunk 4]
```diff
@@ -69,19 +70,8 @@ type Config struct {
...
-	}
-	stepconf.Print(c)
-
-	commits, err := releaseCommits(c.WorkDir, c.NewVersion)
-	if err != nil {
-		failf("Failed to get release commits, error: %v", err)
-	}
-
-	changelog, err := changelogContent(commits, c.NewVersion)
+func processChangelog(cp changelogPrinter, c Config) {
```
Comment (godrei) [inline main.go:73]: only the main function should exit

File: changelog.go [hunk 5]
```diff
@@ -46,7 +47,7 @@ func changelogContent(commits []git.Commit, version string) (string, error) {
 	}
 
 	var buff bytes.Buffer
-	err = tmpl.Execute(&buff, chlog)
+	err = tmpl.Execute(&buff, cl)
```
Comment (godrei) [inline changelog.go:50]: NOTE: could be moved to the if

File: main.go (hunk 4 above)
Comment (godrei) [inline main.go:73]: it is a bit strange, that you pass Config as it is and pass some of its properties via changelog again

File: main_test.go [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+	return "line 1\nline 2", nil
+}
+
+type mockTooManyChangelogPrinter struct{}
+
+func (mcp *mockTooManyChangelogPrinter) content() (string, error) {
+	return strings.Repeat("a", 25*1024), nil
+}
+
+func Test_processChangelog(t *testing.T) {
```
Comment (godrei) [inline main_test.go:20]: the unit test reveals problems with the processChangelog function: no return value, can not directly assert

**Comment by godrei** [comment]
Agree with @lpusok exporting the changelog path as output would worth it.

File: main.go (hunk 3 above)
Comment (trapacska) [reply main.go:102]: Discussed with the PM and we will not export the file path in the step, as the path is available in other steps and reusable.

**Review by godrei — APPROVED** [review]

**Pull request merged**